### PR TITLE
refactor makefile

### PR DESCRIPTION
**Pull request by anmyachev**
- rewrite `linux.yml` for using `make`
- added possibility to do `make init` with concrete service
- added possibility to run tests for separate backend via `make test BACKENDS=omniscidb`

### CODE REVIEW

**Comment by xmnlab** [comment]
hey @jreback if you have time, could you also take a look into this PR pls?

**Comment by jreback** [comment]
yes

File: Makefile [hunk 1]
```diff
@@ -4,11 +4,47 @@ SHELL := /bin/bash
...
-PYTEST_OPTIONS :=
+DOCKER := PYTHON_VERSION=$(PYTHON_VERSION) docker-compose -f $(COMPOSE_FILE)
+DOCKER_UP := $(DOCKER) up --remove-orphans -d --no-build
+DOCKER_RUN := $(DOCKER) run --rm
+DOCKER_BUILD := $(DOCKER) build
+DOCKER_STOP := $(DOCKER) rm --force --stop
+
+DOCKER_RUN_COMMAND := echo "you should do 'make docker_run DOCKER_RUN_COMMAND=[you command]'"
+
+BACKENDS := clickhouse impala kudu-master kudu-tserver mysql omniscidb parquet postgres sqlite
```
Comment (jreback) [inline Makefile]: can you add some comments here (and elsewhere if needed), and ideally doc this in the development docs, that when a new backend is added this needs to be updated.

File: Makefile [hunk 2]
```diff
@@ -4,11 +4,47 @@ SHELL := /bin/bash
 MAKEFILE_DIR = $(patsubst %/,%,$(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
 PYTHON_VERSION := 3.6
 PYTHONHASHSEED := "random"
+
+# docker specific
 COMPOSE_FILE := "$(MAKEFILE_DIR)/ci/docker-compose.yml"
-DOCKER := docker-compose -f $(COMPOSE_FILE)
-DOCKER_RUN := PYTHON_VERSION=${PYTHON_VERSION} $(DOCKER) run --rm
-PYTEST_OPTIONS :=
+DOCKER := PYTHON_VERSION=$(PYTHON_VERSION) docker-compose -f $(COMPOSE_FILE)
```
Comment (jreback) [inline Makefile]: also liberally add comments in the Makefile for a casual reader to grok

File: ci/backends-markers.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash -e
+
```
Comment (jreback) [inline ci/backends-markers.sh]: can you give some comments on usage here (I know this is called from the other scripts), but nice to have

**Comment by jreback** [comment]
@icexelloss if you can try out and see

**Comment by jreback** [comment]
@anmyachev any idea why the lint step takes 11 min: https://dev.azure.com/ibis-project/ibis/_build/results?buildId=1766&view=logs&jobId=6a991cd6-976c-5315-84e1-201ee118e969&j=0bcbe311-d8da-5917-dfb4-bce063779e14&t=7f67ce4b-1c2a-5dd8-6c81-78d8a43bcf25

**Comment by anmyachev** [comment]
@jreback 
- problem with lint fixed
- some comments added in Makefile
- documentation was update

**Comment by anmyachev** [comment]
@jreback the PR is ready for review

File: Makefile [hunk 4]
```diff
@@ -2,14 +2,56 @@
 
 SHELL := /bin/bash
 MAKEFILE_DIR = $(patsubst %/,%,$(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
-PYTHON_VERSION := 3.6
+PYTHON_VERSION := 3.6 # or you can use '3.7'
```
Comment (jreback) [inline Makefile]: how does this interact with the PYTHON variable in Dockerfile.dev?

can you give some comments that this ultimately installs ci/requirements-PYTHON.yml

Comment (anmyachev) [reply Makefile]: @jreback thanks for review. Comments about PYTHON_VERSION variable were added.

**Comment by anmyachev** [comment]
@jreback the PR is ready for review

File: Makefile [hunk 5]
```diff
@@ -2,14 +2,61 @@
 
 SHELL := /bin/bash
 MAKEFILE_DIR = $(patsubst %/,%,$(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
+
+# from PYTHON_VERSION depends which `./ci/requirements-$PYTHON_VERSION-dev` file
+# will be used for creating ibis image (see for additional info: `./ci/Dockerfile.dev`
```
Comment (jreback) [inline Makefile]: so these comments are fine, but maybe my question wasn't clear. we use the PYTHON variable inside Dockerfile.dev and NOT PYTHON_VERSION. I think we should simply fix this or is there a reason that is not obvious?

**Comment by anmyachev** [comment]
@jreback ping on green

@xmnlab @jreback thanks for reviewing!

**Comment by jreback** [comment]
thanks @anmyachev

**Pull request merged**